core/store/models/common.go

```diff
@@ -425,20 +425,46 @@ func (c Cron) String() string {
 	return string(c)
 }
 
-// Duration is a time duration.
-type Duration time.Duration
+// Duration is a non-negative time duration.
```

Comment:
I cannot believe golang's default allows for negative durations.

What were they thinking? Time travel?

Reply:
They provide a handy way to compute the time some duration before another time, I guess. I added a `Before` method to do that, though.